app/Http/Middleware/CheckScopesForAPI.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ *    Copyright 2015-2018 ppy Pty. Ltd.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace App\Http\Middleware;
+
+use Laravel\Passport\Http\Middleware\CheckScopes;
+
+class CheckScopesForAPI
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure                 $next
+     * @param  mixed                ...$scopes
+     * @return \Illuminate\Http\Response
+     * @throws \Illuminate\Auth\AuthenticationException
+     * @throws \Laravel\Passport\Exceptions\MissingScopeException
+     */
+    public function handle($request, $next, ...$scopes)
```

Comment:
is this good practice? I don't know if middleware are supposed to call other middleware

Reply:
You might as well just use `Laravel\Passport\Http\Middleware\CheckScopes::class` instead of this file, since you'll only need to be defining the scopes on API routes anyways.

Reply:
they're defined on the controllers, not the routes (see conversation above), and most of these controllers are not specific to the API

Reply:
I see, if that's the route they are willing to take, then there isn't really a choice. So by their standard, it's fine.